lens/private/list/flatten.scrbl

```diff
@@ -0,0 +1,63 @@
+#lang scribble/manual
+
+@(require "../doc-util/main.rkt")
+
+@title{Flattening and unflattening lists}
+
+@defthing[append*-lens lens?]{
+A lens that flattens a list one-level down when viewing, and restores
```

Comment:
You're right, I should explain this. (And probably also write versions of `append*` and `restore-structure` that check the types of their arguments)